### ISSUE TITLE
Feature Request: Add option to disable other tplure when leashed by owner

### ISSUE THREAD

**Issue by SimplyNemz** [OpenCollarTeam/OpenCollar#310]
Hey all! I'm not a scripter, can't say how easy this would be, but after poking around figured it wouldn't be _that_ difficult.

If you are leashed to someone you would expect to, you know, stay with them right? But if you have multiple owners on your collar or someone uses RLV capture options you can be pulled away from your leash point. You might bounce back if you're close enough, but when you're pulled a sim or two away.. The person who had you leashed might rightly be angry. Specially if you have strict leash activated, have already been releashed, and now can't teleport back. Oh, and for those bored this is totally how you can play hot potato with a shared submissive.

So my thought is adding an additional block that's added as soon as you are leashed. Found a code idea that could be modified to work? Again, not scripter, no idea if this can actually be done. http://wiki.secondlife.com/wiki/User:Toy_Wylie/RLV_Documentation/tplure_sec

TLDR: Add command on leashing that prevents others/devices from teleporting you away.

**Comment by zontreck** [comment]
I'm 90% sure this is already a feature.

**Comment by zontreck** [comment]
Just confirmed it. If you turn off strict leash, tplure will not be issued among other commands.

**Comment by zontreck** [comment]
If this isn't what you are looking for please reply and elaborate on what it is you're requesting exactly

**Comment by SimplyNemz** [comment]
I may have had the wrong command. =P

Basically, want to be able to set that when leashed, no one else can use RLV to teleport you away.

Say I have two owners.
Owner 1 has me leashed.
Owner 2 tries to teleport me.

Right now Owner 2 WOULD teleport me. What would make more sense though is if Owner two gets a message saying I am currently leashed and unavailable.

**Comment by SharieCriss** [comment]
Unfortunately the way that tplure is implemented in the viewer's R:V code, your RLV item (such as the collar) has no idea that a request is coming in - it either is allowed or not allowed by the viewer based on the restrictions. Probably the best way to handle it is something like:
If strict-leashed by an owner, remove the tplure exceptions for other owners. When unleashed, re-enable exceptions for other owners. Would this satisfy your concerns?

**Comment by zontreck** [comment]
Keep in mind that while what sharie suggests would be possible, it would only be possible if leashed by cmd_owner, any other Auth level wouldnt remove exceptions

**Comment by SharieCriss** [comment]
Yes, that's exactly what my thoughts were, I look at it as Owners have priority, but in my example, you wouldn't get yanked from one owner to another at least. It would allow an owner to retrieve you from a non-owner. The the secure version of TPLure still obeys the exceptions from the same device so won't stop an owner from TPing you no matter what. The only way to stop this behavior is to remove the exception. Limiting the exception removal to only happen when an owner has you leashed is probably one of the least objectionable cases.

**Comment by SimplyNemz** [comment]
I would say to maybe not make it part of 'Strict Leash' but a separate option? More flexibility with enforcement options that way.

And yeah, agree with Sharie. It would make sense to that it would ONLY work with an owner. As an owner should always have the right to steal away their slave/pet/property/thing. But at least it would prevent one owner from taking you from another owner.

**Comment by zontreck** [comment]
I changed the ticket title to be more clear in what is being requested. Also added the help wanted tag incase anyone feels like tackling this and adding it in. The scripts that would need modifications are oc_leash, and oc_rlvextension.

**Comment by zontreck** [comment]
Removing from 8.0 project queue. This will be considered for a future version
